### PR TITLE
AB#95917 - API: add survey general settings endpoints(SDK)

### PR DESCRIPTION
**Pull request by eshafarevich-sc**
API: add survey general settings endpoints
[AB#95917 ](https://dev.azure.com/niposoftware/Nfield/_workitems/edit/95917)

Main Changes:
- New SurveyGeneralSettings service
- Postman collection added for the new service.

PRs:
- https://github.com/NIPOSoftware/Nfield-SDK/pull/269
- https://github.com/NIPOSoftwareBV/nfield-source/pull/6939

Tests - go to: https://github.com/NIPOSoftwareBV/nfield-source/pull/6939

### CODE REVIEW

File: Library/Services/INfieldSurveyGeneralSettingsService.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+        /// <summary>
+        /// Updates the supplied <paramref name="GeneralSetting"/>
+        /// </summary>
+        /// <param name="GeneralSetting">The <see cref="SurveyGeneralSetting"/> to update</param>
+        /// <exception cref="T:System.AggregateException"></exception>
+        /// The aggregate exception can contain:
+        /// <exception cref="Nfield.Exceptions.NfieldErrorException"></exception>
+        /// <exception cref="Nfield.Exceptions.NfieldHttpResponseException"></exception>
+        Task UpdateAsync(string surveyId, SurveyGeneralSetting GeneralSetting);
```
Comment (ozaragoza) [inline Library/Services/INfieldSurveyGeneralSettingsService.cs:44]:
```suggestion
        Task UpdateAsync(string surveyId, SurveyGeneralSetting generalSettings);
```

File: Library/Models/SurveyGeneralSetting.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Holds the properties of a survey general setting
+    /// </summary>
+    public class SurveyGeneralSetting
```
Comment (ozaragoza) [inline Library/Models/SurveyGeneralSetting.cs:21]:
```suggestion
    public class SurveyGeneralSettings
```

File: Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                         .ContinueWith(
+                             stringTask =>
+                             JsonConvert.DeserializeObject<SurveyGeneralSetting>(stringTask.Result))
+                         .FlattenExceptions();
+        }
+
+        /// <summary>
+        /// See <see cref="INfieldSurveyGeneralSettingsService.PatchAsync"/>
+        /// </summary>
+        public Task UpdateAsync(string surveyId, SurveyGeneralSetting models)
```
Comment (ozaragoza) [inline Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs:54]:
```suggestion
        public Task UpdateAsync(string surveyId, SurveyGeneralSetting generalSettings)
```

File: Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        private Uri SurveyGeneralSettingsApi(string surveyId)
+        {
+            return new Uri(ConnectionClient.NfieldServerUri, $"SurveyGeneralSettings/{surveyId}");
+        }
+
+        private INfieldHttpClient Client => ConnectionClient.Client;
+
+        private static void CheckSurveyId(string surveyId)
+        {
+            if (surveyId == null)
```
Comment (ozaragoza) [inline Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs:90]: Better if you use string.IsNullOrEmpty(...)

File: Library/Services/INfieldSurveyGeneralSettingsService.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+{
+
+    /// <summary>
+    /// Represents a set of methods to get and update  <see cref="SurveyGeneralSetting"/>.
+    /// </summary>
+    public interface INfieldSurveyGeneralSettingsService
+    {
+
+        /// <summary>
+        /// This method returns SurveyGeneralSetting status
```
Comment (kefishev) [inline Library/Services/INfieldSurveyGeneralSettingsService.cs:31]: status?

File: Library/Services/INfieldSurveyGeneralSettingsService.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Nfield.Models;
```
Comment (kefishev) [inline Library/Services/INfieldSurveyGeneralSettingsService.cs:19]: Clean up

File: Library/Services/INfieldSurveyGeneralSettingsService.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        /// <summary>
+        /// This method returns SurveyGeneralSetting status
+        /// </summary>
+        /// <param name="surveyId">The id of the survey to get the SurveyGeneralSetting</param>
+        Task<SurveyGeneralSettings> QueryAsync(string surveyId);
+
+        /// <summary>
+        /// Updates the supplied <paramref name="GeneralSetting"/>
+        /// </summary>
+        /// <param name="GeneralSetting">The <see cref="SurveyGeneralSetting"/> to update</param>
```
Comment (kefishev) [inline Library/Services/INfieldSurveyGeneralSettingsService.cs:39]: GeneralSetting -> generalSetting

File: Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Nfield.Extensions;
+using Nfield.Infrastructure;
+using Nfield.Models;
+
```
Comment (kefishev) [inline Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs:24]: Clean up

File: Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                         .ContinueWith(
+                             stringTask =>
+                             JsonConvert.DeserializeObject<SurveyGeneralSettings>(stringTask.Result))
+                         .FlattenExceptions();
+        }
+
+        /// <summary>
+        /// See <see cref="INfieldSurveyGeneralSettingsService.PatchAsync"/>
+        /// </summary>
+        public Task UpdateAsync(string surveyId, SurveyGeneralSettings generalSettings)
```
Comment (kefishev) [inline Library/Services/Implementation/NfieldSurveyGeneralSettingsService.cs:54]: generalSettings doesn't much interface declaration

File: Tests/Services/NfieldSurveyGeneralSettingsServiceTests.cs [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Moq;
+using Newtonsoft.Json;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Services.Implementation;
+using Xunit;
+
```
Comment (kefishev) [inline Tests/Services/NfieldSurveyGeneralSettingsServiceTests.cs:27]: clean up

File: Tests/Services/NfieldSurveyGeneralSettingsServiceTests.cs [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+namespace Nfield.Services
+{
+    /// <summary>
+    /// Tests for <see cref="NfieldSurveyGeneralSettingsService"/>
+    /// </summary>
+    public class NfieldSurveyGeneralSettingsServiceTests : NfieldServiceTestsBase
+    {
+        private readonly NfieldSurveyGeneralSettingsService _target;
+        readonly Mock<INfieldHttpClient> _mockedHttpClient;
```
Comment (kefishev) [inline Tests/Services/NfieldSurveyGeneralSettingsServiceTests.cs:36]: private ?

File: Library/Services/INfieldSurveyGeneralSettingsService.cs [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        /// <summary>
+        /// This method returns SurveyGeneralSetting
+        /// </summary>
+        /// <param name="surveyId">The id of the survey to get the SurveyGeneralSetting</param>
+        Task<SurveyGeneralSettings> QueryAsync(string surveyId);
+
+        /// <summary>
+        /// Updates the supplied <paramref name="GeneralSetting"/>
+        /// </summary>
+        /// <param name="generalSetting">The <see cref="surveyGeneralSetting"/> to update</param>
```
Comment (kefishev) [inline Library/Services/INfieldSurveyGeneralSettingsService.cs:36]: The cross reference is still wrong, it is also wrong at line 23

**Review by kefishev — APPROVED** [review]

**Pull request merged**